Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -318,8 +269,10 @@ google_firestore_v1_Value Serializer::EncodeFieldValue(
       return result;
 
     case FieldValue::Type::Reference:
-      // TODO(rsgowman): Implement
-      abort();
+      result.which_value_type = google_firestore_v1_Value_reference_value_tag;
```

Comment:
One thing I like about the Objective-C serializer is that most of the per-type encoding/decoding is done in separate methods that can be addressed directly. This would have the effect of making the addition for referencevalue look like this:

```c++
google_firestore_v1_Value EncodeReferenceValue(const FieldValue::Reference& ref) const;
```

And it would make this caller look like this:

```c++
  case FieldValue::Type::Reference:
    return EncodeReferenceValue(field_value.reference_value());
```

What do you think of this pattern? I'm wary of having this really large methods with lots of different logic for all the different types.

Reply:
This doesn't seem to have changed?